packages/now-build-utils/src/detect-builders.ts

```diff
@@ -83,6 +83,11 @@ function getApiFunctionBuilder(
     });
   }
 
+  const { includeFiles, excludeFiles } = fn;
+
+  if (includeFiles) Object.assign(config, { includeFiles });
+  if (excludeFiles) Object.assign(config, { excludeFiles });
```

Comment:
This doesn't seem right because it sounds like you want to continue using the implementation in the Builder rather than implement our own include/exclude logic.

I think it would be best to remove the logic from the builder and instead add additional files to the lambda outside the builder.

Reply:
We've thought about that, but we don't want to tackle it for now. For several reasons:

- It will take quite a while to extensively test that logic and make it reliable
- We can't be sure the customer is using a bundler that might need to know which files to include or exclude (like ncc)

Reply:
@styfle yes, we keep it like that for now. Take a look at one of the last comments of [PRODUCT-27].

Reply:
> It will take quite a while to extensively test that logic and make it reliable

I think it would be easier to test because you wouldn't need separate tests for each builder like we have now.

> We can't be sure the customer is using a bundler that might need to know which files to include or exclude (like ncc)

I'm not sure how this would make a difference. The idea is that the user is _overriding_ whatever bundling mechanism may exist in the builder. They're saying "I know better, just dump these files in the lamdba".

Reply:
@styfle That works for including, but not for excluding. If I want to prevent a bundler from excluding the file, I need to explicitly tell it. The only other possibility would be to strip it from fargate so that the bundler can't find it - but that would throw an error.

Reply:
> I think it would be easier to test because you wouldn't need separate tests for each builder like we have now.

This is a great point, however.